### PR TITLE
Psr3FormatterTest

### PR DESCRIPTION
**Pull request by daweeb**
Added Test:

- Exception test for Constructor
- Format test with Valid and invalid text Context

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -39,7 +39,8 @@
     "require-dev": {
         "horde/test": "^3 || dev-FRAMEWORK_6_0",
         "horde/cli": "^3 || dev-FRAMEWORK_6_0",
-        "horde/scribe": "^3 || dev-FRAMEWORK_6_0"
+        "horde/scribe": "^3 || dev-FRAMEWORK_6_0",
+        "phpunit/phpunit": "10.0.x-dev"
```
Comment (midahp) [inline composer.json]: was this intentional or by accident?
The horde/test requirement should already install phpunit9. Do we need phpunit10 for anything?

File: test/Formatter/Psr3FormatterTest.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+{
+    public function SetUp(): void
+    {
+        $this->messageLog = new LogLevel(3, 'info');
+        $this->message = new LogMessage($this->messageLog, '{datum} everything not saved, will be lost {name}');
+    }
+
+    public function testFormatterReplacesMessageWithContext()
+    {
+        $this->formatter = new Psr3Formatter(['name' => 'Voldemort','datum' => '1550071894']);
```
Comment (midahp) [inline test/Formatter/Psr3FormatterTest.php:33]: you dont need to save the formatter as a class attribute here.
Also applies to other lines. Not a problem, but not needed.

File: test/Formatter/Psr3FormatterTest.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        $formatted = $this->message->formatMessage([$this->formatter]);
+        $this->assertIsString($formatted);
+        $this->assertStringContainsString('Voldemort', $formatted);
+        $this->assertStringContainsString('1550071894', $formatted);
+
+        $this->formatter = new Psr3Formatter([]);
+        $formatted = $this->message->formatMessage([$this->formatter]);
+        $this->assertIsString($formatted);
+        $this->assertEquals('{datum} everything not saved, will be lost {name}', $formatted);
```
Comment (midahp) [inline test/Formatter/Psr3FormatterTest.php:42]: This should maybe be a separate test method

**Review by midahp — COMMENTED** [review]
What is missing is a test that checks if the formatter also replaces entries returned from the LogMessage `context` method.

File: composer.json (hunk 1 above)
Comment (ralflang) [reply composer.json]: Neither should. A package should require-dev on horde/test and make no assumption if phpunit is installed to the project or called using an external phar.

Comment (daweeb) [reply composer.json]: @ralflang @midahp falsely added, correct one in usr/local/bin in use

**Pull request merged**